src/smt/theory_str_noodler/nielsen_decision_procedure.cpp

```diff
@@ -0,0 +1,567 @@
+#include <queue>
+#include <utility>
+#include <algorithm>
+
+#include <mata/nfa-strings.hh>
+#include "util.h"
+#include "aut_assignment.h"
+#include "nielsen_decision_procedure.h"
+
+namespace smt::noodler {
+
+
+    /**
+     * @brief Get length constraints of the solution
+     *
+     * @param variable_map Mapping of BasicTerm variables to the corresponding z3 variables
+     * @return expr_ref Length formula describing all solutions (currently true)
+     */
+    expr_ref NielsenDecisionProcedure::get_lengths(const std::map<BasicTerm, expr_ref>& variable_map) {
+        if(this->init_length_sensitive_vars.size() == 0) {
+            return expr_ref(m.mk_true(), m);
+        }
+
+        std::map<BasicTerm, expr_ref> actual_var_map{};
+        expr_ref formula(this->m.mk_true(), this->m);
+        for(size_t i = 0; i < this->length_paths.size(); i++) {
+            if(this->length_paths_index >= this->length_paths[i].size()) {
+                continue;
+            }
+            expr_ref part_fle = length_formula_path(this->length_paths[i][this->length_paths_index], variable_map, actual_var_map);
+            formula = this->m.mk_and(formula, part_fle);
+        }
+        formula = this->m.mk_and(formula, generate_len_connection(variable_map, actual_var_map));
+
+        STRACE("str", tout << "Nielsen lengths: " << mk_pp(formula, this->m) << std::endl;);
+        this->length_paths_index++;
+        return formula;
+    }
+
+    /**
+     * @brief Initialize computation.
+     */
+    void NielsenDecisionProcedure::init_computation() {
+
+    }
+
```

Comment:
Should something be here? It just looks suspicious that we have nothing to initialize here.